profondo/cnn.py

```diff
@@ -124,11 +136,27 @@ def report_callback(epoch, logs):
         print(model.summary())
         callbacks.append(LambdaCallback(on_epoch_end=report_callback))
 
-    # TODO: use ImageDataGenerator to augment images
-    model.fit(x_train, y_train,
-              batch_size=args.batch_size,
-              epochs=args.epochs,
-              verbose=args.logging,
-              validation_data=(x_test, y_test),  # TODO: don't use test data
-              class_weight=genre_weights,
-              callbacks=callbacks)
+    if args.augment:
+        # Use ImageDataGenerator to augment images
+        datagen = ImageDataGenerator(
+            featurewise_center=True,
+            featurewise_std_normalization=False,
+            horizontal_flip=True,
+            vertical_flip=True)
+        datagen.fit(x_train)
+
+        model.fit_generator(datagen.flow(x_train, y_train, batch_size=args.batch_size),
+                  steps_per_epoch=2*args.batch_size,
```

Comment:
My mistake, this should be: "It should typically be equal to the number of unique samples of your dataset divided by the batch size." from [here](https://keras.io/models/sequential/)